### DISCUSSION TITLE
[Bug report] Upload of a new site is stalling at ~50%, then produces an error message

### DISCUSSION THREAD

**Discussion by OttoKerner** [GetPublii/Publii#1631] [Bug Report]
### Operating system

Windows 11 23H2

### Publii version

0.46.1 (build 16918)

### Issue type

Application

### Bug description

During the initial upload of a new website, the progress bar is stalling at 234 of 553 operations.
I've let it sit for a few minutes, then Aborted the process. 
I was then greeted by an error modal 
![grafik](https://github.com/user-attachments/assets/32221acd-be1e-413b-8b88-75ad71ba16db)
Repeated attempts always stall at the operation #234

### Steps to reproduce

Upload a new small-ish website per FTP. 
I'm unsure how _not_ to reproduce, as its the first time I am using this feature.

### Relevant log output

```shell
[Fri, 23 Aug 2024 13:05:05 GMT] [connection] < '227 Entering Passive Mode (92,204,55,224,253,12).\r\n'
[Fri, 23 Aug 2024 13:05:05 GMT] [parser] < '227 Entering Passive Mode (92,204,55,224,253,12).\r\n'
[Fri, 23 Aug 2024 13:05:05 GMT] [parser] Response: code=227, buffer='Entering Passive Mode (92,204,55,224,253,12).'
[Fri, 23 Aug 2024 13:05:05 GMT] [connection] PASV socket connected
[Fri, 23 Aug 2024 13:05:05 GMT] [connection] > 'STOR test/assets/css/photoswipe.css'
[Fri, 23 Aug 2024 13:05:05 GMT] FTP ERROR: Error: read ECONNRESET
[Fri, 23 Aug 2024 13:05:05 GMT] FTP CONNECTION CLOSED: false
```

**Comment by OttoKerner** [comment]
A manual upload of the website catalog was successful using a regular FTP client, on the same computer.
But that is unfortunately no solution, as I'm evaluating this for use by a rather non-technical user.

**Comment by OttoKerner** [comment]
After the manual upload, subsequent attempts were successful. I assume because the number of operations was lower.

**Comment by founfabug** [comment]
I can confirm this behavior during a site update (after prior successful upload and updates), and including the correct operation after manual upload. It does seem related to the complexity of the upload, in my case around 10 single paragraph posts with featured images.

**Comment by TheRealBeef** [comment]
I am encountering a similar issue with FTPS from Ubuntu 24.10 with Version: 0.46.3 (build 17008) of Publii.

It stalls on operation 22 of 77 for updating the website, although I'm not getting any error messages even after waiting 15 minutes before cancelling.

**Reply by TheRealBeef** [reply]
Solved by following the steps to enable the experimental FTP library at - https://github.com/GetPublii/Publii/issues/1417#issuecomment-2542450228

**Comment by bjazmoore** [comment]
Perhaps Publii needs to set the keep-alive option in the connection.  It seems to be timing out.

**Comment by MartinCowen** [comment]
I'm getting the same problem, stalling at 226/1242, using FTPS on Publii 0.47.1 (build 17302) on Windows 11 24H2. Resolved using Windows Explorer into the FTP server to manually copy the files from the output folder, then there's nothing to sync so succeeds. 136 MB, 1521 files.

**Reply by dziudek** [reply]
@MartinCowen - do you have the same issue if you enable "Use alternative FTP library for deployment" under App Settings?
